R package install does not work for clang

Hi - I get an error when I try to install the R package.  It may be a clang vs gcc issue, but I'm not really a C++ programmer so I'm not sure. I'm using clang6, which is now recommended by CRAN for R builds on macOS (https://cran.r-project.org/bin/macosx/tools/).  Thanks in advance!

My output is:

R CMD build libfrn-R
R CMD INSTALL RFN_0.1.tar.gz

R CMD INSTALL RFN_0.1.tar.gz
* installing to library ‘/Library/Frameworks/R.framework/Versions/3.5/Resources/library’
* installing *source* package ‘RFN’ ...
checking for /opt/intel/mkl/include/mkl_blas.h... no
configure: creating ./config.status
config.status: creating src/Makevars
** libs
PKG_LIBS is     
BL_LIBS is   
/usr/local/clang6/bin/clang++  -Wall -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -DCOMPILE_FOR_R -I -I/usr/local/include -DNOGPU -I"/Library/Frameworks/R.framework/Versions/3.5/Resources/library/Rcpp/include" -I/usr/local/include  -std=c++11 -fPIC  -Wall -g -O2  -c nist_spblas.cc -o nist_spblas.o
nist_spblas.cc:390:21: error: invalid operands to binary expression ('double'
      and 'complex<double>')
       y[p->second] += alpha * conj(p->first);
       ~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~~~~
nist_spblas.cc:481:7: note: in instantiation of member function
      'NIST_SPBLAS::TSp_mat<double>::sp_conj_axpy' requested here
      sp_conj_axpy( alpha * *X, S[i], y, incy);
      ^
nist_spblas.cc:515:5: note: in instantiation of member function
      'NIST_SPBLAS::TSp_mat<double>::nondiag_mult_vec_conj_transpose' requested
      here
    nondiag_mult_vec_conj_transpose(alpha, x, incx, y, incy);
    ^
nist_spblas.cc:840:5: note: in instantiation of member function
      'NIST_SPBLAS::TSp_mat<double>::mult_vec_conj_transpose' requested here
    mult_vec_conj_transpose(alpha, x, incx, y, incy);
    ^
nist_spblas.cc:1711:13: note: in instantiation of member function
      'NIST_SPBLAS::TSp_mat<double>::usmv' requested here
  return M->usmv(transa, alpha, x, incx, y, incy);
            ^
nist_spblas.cc:1928:10: note: in instantiation of function template
      specialization 'BLAS_xusmv<double>' requested here
  return BLAS_xusmv(
         ^
nist_spblas.cc:421:10: error: invalid operands to binary expression ('double'
      and 'complex<double>')
      *Y += alpha * conj(*d) * *X;
      ~~ ^  ~~~~~~~~~~~~~~~~~~~~~
nist_spblas.cc:518:7: note: in instantiation of member function
      'NIST_SPBLAS::TSp_mat<double>::mult_conj_diag' requested here
      mult_conj_diag(alpha, x, incx, y, incy);
      ^
nist_spblas.cc:840:5: note: in instantiation of member function
      'NIST_SPBLAS::TSp_mat<double>::mult_vec_conj_transpose' requested here
    mult_vec_conj_transpose(alpha, x, incx, y, incy);
    ^
nist_spblas.cc:1711:13: note: in instantiation of member function
      'NIST_SPBLAS::TSp_mat<double>::usmv' requested here
  return M->usmv(transa, alpha, x, incx, y, incy);
            ^
nist_spblas.cc:1928:10: note: in instantiation of function template
      specialization 'BLAS_xusmv<double>' requested here
  return BLAS_xusmv(
         ^
nist_spblas.cc:348:17: error: invalid operands to binary expression ('double'
      and 'complex<double>')
            sum += conj(p->first) * x[p->second];
            ~~~ ^  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
nist_spblas.cc:453:25: note: in instantiation of member function
      'NIST_SPBLAS::TSp_mat<double>::sp_conj_dot_product' requested here
        y[i] += alpha * sp_conj_dot_product(S[i], x, incx);
                        ^
nist_spblas.cc:521:7: note: in instantiation of member function
      'NIST_SPBLAS::TSp_mat<double>::nondiag_mult_vec_conj' requested here
      nondiag_mult_vec_conj(alpha, x, incx, y, incy);
      ^
nist_spblas.cc:840:5: note: in instantiation of member function
      'NIST_SPBLAS::TSp_mat<double>::mult_vec_conj_transpose' requested here
    mult_vec_conj_transpose(alpha, x, incx, y, incy);
    ^
nist_spblas.cc:1711:13: note: in instantiation of member function
      'NIST_SPBLAS::TSp_mat<double>::usmv' requested here
  return M->usmv(transa, alpha, x, incx, y, incy);
            ^
nist_spblas.cc:1928:10: note: in instantiation of function template
      specialization 'BLAS_xusmv<double>' requested here
  return BLAS_xusmv(
         ^
nist_spblas.cc:627:15: error: invalid operands to binary expression ('double'
      and 'typename __libcpp_complex_overload_traits<double>::_ComplexType'
      (aka 'complex<double>'))
        x[jj] /= conj(diag[j]) ;
        ~~~~~ ^  ~~~~~~~~~~~~~
nist_spblas.cc:885:16: note: in instantiation of member function
      'NIST_SPBLAS::TSp_mat<double>::transpose_triangular_conj_solve' requested
      here
        return transpose_triangular_conj_solve(alpha, x, incx);
               ^
nist_spblas.cc:1737:13: note: in instantiation of member function
      'NIST_SPBLAS::TSp_mat<double>::ussv' requested here
  return M->ussv(transa, alpha, x, incx);
            ^
nist_spblas.cc:1959:10: note: in instantiation of function template
      specialization 'BLAS_xussv<double>' requested here
  return BLAS_xussv( transa,
         ^
nist_spblas.cc:390:21: error: invalid operands to binary expression ('float' and
      'complex<float>')
       y[p->second] += alpha * conj(p->first);
       ~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~~~~
nist_spblas.cc:481:7: note: in instantiation of member function
      'NIST_SPBLAS::TSp_mat<float>::sp_conj_axpy' requested here
      sp_conj_axpy( alpha * *X, S[i], y, incy);
      ^
nist_spblas.cc:515:5: note: in instantiation of member function
      'NIST_SPBLAS::TSp_mat<float>::nondiag_mult_vec_conj_transpose' requested
      here
    nondiag_mult_vec_conj_transpose(alpha, x, incx, y, incy);
    ^
nist_spblas.cc:840:5: note: in instantiation of member function
      'NIST_SPBLAS::TSp_mat<float>::mult_vec_conj_transpose' requested here
    mult_vec_conj_transpose(alpha, x, incx, y, incy);
    ^
nist_spblas.cc:1711:13: note: in instantiation of member function
      'NIST_SPBLAS::TSp_mat<float>::usmv' requested here
  return M->usmv(transa, alpha, x, incx, y, incy);
            ^
nist_spblas.cc:2386:10: note: in instantiation of function template
      specialization 'BLAS_xusmv<float>' requested here
  return BLAS_xusmv(
         ^
nist_spblas.cc:421:10: error: invalid operands to binary expression ('float' and
      'complex<float>')
      *Y += alpha * conj(*d) * *X;
      ~~ ^  ~~~~~~~~~~~~~~~~~~~~~
nist_spblas.cc:518:7: note: in instantiation of member function
      'NIST_SPBLAS::TSp_mat<float>::mult_conj_diag' requested here
      mult_conj_diag(alpha, x, incx, y, incy);
      ^
nist_spblas.cc:840:5: note: in instantiation of member function
      'NIST_SPBLAS::TSp_mat<float>::mult_vec_conj_transpose' requested here
    mult_vec_conj_transpose(alpha, x, incx, y, incy);
    ^
nist_spblas.cc:1711:13: note: in instantiation of member function
      'NIST_SPBLAS::TSp_mat<float>::usmv' requested here
  return M->usmv(transa, alpha, x, incx, y, incy);
            ^
nist_spblas.cc:2386:10: note: in instantiation of function template
      specialization 'BLAS_xusmv<float>' requested here
  return BLAS_xusmv(
         ^
nist_spblas.cc:348:17: error: invalid operands to binary expression ('float' and
      'complex<float>')
            sum += conj(p->first) * x[p->second];
            ~~~ ^  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
nist_spblas.cc:453:25: note: in instantiation of member function
      'NIST_SPBLAS::TSp_mat<float>::sp_conj_dot_product' requested here
        y[i] += alpha * sp_conj_dot_product(S[i], x, incx);
                        ^
nist_spblas.cc:521:7: note: in instantiation of member function
      'NIST_SPBLAS::TSp_mat<float>::nondiag_mult_vec_conj' requested here
      nondiag_mult_vec_conj(alpha, x, incx, y, incy);
      ^
nist_spblas.cc:840:5: note: in instantiation of member function
      'NIST_SPBLAS::TSp_mat<float>::mult_vec_conj_transpose' requested here
    mult_vec_conj_transpose(alpha, x, incx, y, incy);
    ^
nist_spblas.cc:1711:13: note: in instantiation of member function
      'NIST_SPBLAS::TSp_mat<float>::usmv' requested here
  return M->usmv(transa, alpha, x, incx, y, incy);
            ^
nist_spblas.cc:2386:10: note: in instantiation of function template
      specialization 'BLAS_xusmv<float>' requested here
  return BLAS_xusmv(
         ^
nist_spblas.cc:627:15: error: invalid operands to binary expression ('float' and
      'typename __libcpp_complex_overload_traits<float>::_ComplexType'
      (aka 'complex<float>'))
        x[jj] /= conj(diag[j]) ;
        ~~~~~ ^  ~~~~~~~~~~~~~
nist_spblas.cc:885:16: note: in instantiation of member function
      'NIST_SPBLAS::TSp_mat<float>::transpose_triangular_conj_solve' requested
      here
        return transpose_triangular_conj_solve(alpha, x, incx);
               ^
nist_spblas.cc:1737:13: note: in instantiation of member function
      'NIST_SPBLAS::TSp_mat<float>::ussv' requested here
  return M->ussv(transa, alpha, x, incx);
            ^
nist_spblas.cc:2417:10: note: in instantiation of function template
      specialization 'BLAS_xussv<float>' requested here
  return BLAS_xussv( transa,
         ^
nist_spblas.cc:106:9: warning: private field 'general_' is not used
      [-Wunused-private-field]
    int general_;
        ^
1 warning and 8 errors generated.
make: *** [nist_spblas.o] Error 1
ERROR: compilation failed for package ‘RFN’

--- a/nist_spblas.cc
+++ b/nist_spblas.cc
@@ -345,7 +345,7 @@ class TSp_mat : public Sp_mat
           for ( typename vector< pair<T,int> >::const_iterator p = r.begin();
             p < r.end(); p++)
           {
-            sum += conj(p->first) * x[p->second];
+            sum += (p->first) * x[p->second];
           }
         }
         else /* incx != 1 */
@@ -354,7 +354,7 @@ class TSp_mat : public Sp_mat
             p < r.end(); p++)
            {
             //sum = sum + p->first * x[p->second * incx];
-            sum += conj(p->first) * x[p->second * incx];
+            sum += (p->first) * x[p->second * incx];
             }
         }
 
@@ -387,14 +387,14 @@ class TSp_mat : public Sp_mat
     {
       for (typename vector< pair<T,int> >::const_iterator p = r.begin();
           p < r.end(); p++)
-       y[p->second] += alpha * conj(p->first);
+       y[p->second] += alpha * p->first;
 
     }
     else /* incy != 1 */
     {
     for (typename vector< pair<T,int> >::const_iterator p = r.begin();
         p < r.end(); p++)
-      y[incy * p->second] += alpha * conj(p->first);
+      y[incy * p->second] += alpha * p->first;
     }
   }
 
@@ -418,7 +418,7 @@ class TSp_mat : public Sp_mat
     typename vector<T>::const_iterator d= diag.begin();
     for ( ; d < diag.end(); X+=incx, d++, Y+=incy)
     {
-      *Y += alpha * conj(*d) * *X;
+      *Y += alpha * (*d) * *X;
     }
   }
 
@@ -624,7 +624,7 @@ class TSp_mat : public Sp_mat
 
       for (int j=N-1, jj=(N-1)*incx; 0<=j; j--, jj -= incx)
       {
-        x[jj] /= conj(diag[j]) ;
+        x[jj] /= (diag[j]) ;
         sp_conj_axpy( -x[jj], S[j], x, incx);
       }
       if (alpha != (T) 1.0)
@@ -638,7 +638,7 @@ class TSp_mat : public Sp_mat
 
       for (int j=0, jj=0; j<N; j++, jj += incx)
       {
-        x[jj] /= conj(diag[j]);
+        x[jj] /= (diag[j]);
         sp_conj_axpy(- x[jj], S[j], x, incx);
       }
       if (alpha != (T) 1.0)
@@ -1581,7 +1581,7 @@ void BLAS_xusdot( enum blas_conj_type conj_flag, int nz,
   }
   else
     for (int i=0; i<nz; i++)
-      t += conj(x[i]) * y[index[i]*incy];
+      t += (x[i]) * y[index[i]*incy];
 
 
   *r = t;

